test/liquid_voting_web/absinthe/mutations/create_delegation/existing_delegations_test.exs

```diff
@@ -1,19 +1,28 @@
 defmodule LiquidVotingWeb.Absinthe.Mutations.CreateDelegation.ExistingDelegationsTest do
   use LiquidVotingWeb.ConnCase
+  import LiquidVoting.Factory
 
   alias LiquidVotingWeb.Schema.Schema
 
-  @delegator_email "delegator@email.com"
-  @delegate_email "delegate@email.com"
-  @another_delegate_email "another-delegate@email.com"
-  @proposal_url "https://www.proposal.com/my"
-  @organization_id Ecto.UUID.generate()
+  describe "create global delegation when a global delegation for a different delegate already exists" do
+    setup do
+      global_delegation = insert(:delegation)
+      another_delegate = insert(:participant)
 
-  describe "with existing global delegation" do
-    test "overwrites existing global delegation" do
+      [
+        organization_id: global_delegation.organization_id,
+        global_delegation_id: global_delegation.id,
+        delegator: global_delegation.delegator,
+        another_delegate: another_delegate
+      ]
+    end
+
+    test "overwrites existing global delegation", context do
       query = """
       mutation {
-        createDelegation(delegatorEmail: "#{@delegator_email}", delegateEmail: "#{@delegate_email}") {
+        createDelegation(delegatorEmail: "#{context[:delegator].email}", delegateEmail: "#{
```

Comment:
Sorry, I meant replacing the test setup that's now done through absinthe requests, like this request to create the existing global delegation, with factory creation.

Test setup as in test data state preparation, and not [ExUnit's setup callback](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html)

Reply:
OK, but I don't quite understand what that would look like. The docs you link don't have absinthe examples, and I can't find any online. I want to know what using the ExUnit callback for a test setup looks like.

Reply:
...just noticed this test actually only has one request :) My comment applies to the next test on line 70, for example.

In other words, let's not use ExUnit's setup, move the factories inline in each test, and replace setup _requests_ with factories.

I hope I'm making sense =D

Reply:
I'm sure you are making sense, but I don't understand :disappointed:

Reply:
But I probably would, given an example.

Reply:
Update: we hashed it out on a slack call. Just need to remove setup blocks and inline factories, and then remove extra request assertions that are already tested at the unit leve.

Reply:
@oliverbarnes I think I've got it now :blush: